include/zephyr/drivers/dma/dma_rpi_pico.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2023 TOKITA Hiroshi <tokita.hiroshi@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_DMA_RPI_PICO_H_
+#define ZEPHYR_INCLUDE_DRIVERS_DMA_RPI_PICO_H_
+
+#include <zephyr/device.h>
+#include <zephyr/dt-bindings/dma/rpi_pico_dma.h>
+
+extern int rpi_pico_dma_channel_set_high_priority(const struct device *dev, uint32_t ch, bool hpri);
```

Comment:
Why are these functions/file needed? And why is the `extern` needed?

Reply:
It defines rpi_pico_dma specific APIs.
That is not fit in zephyr's common API features.

Reply:
There is a channel_priority field in struct dma_config which could almost certainly be used for this

Reply:
I recently updated the rpi_pico's HAL module. This includes the DMA priority API.
I updated this PR to use that API with refer to the `channel_priority` value.
The rpi_pico specific function is no needed.
I removed it.